### ISSUE TITLE
Not compatible with PHP 8.0

### ISSUE THREAD

**Issue by fvicent** [CleanTalk/mybb-antispam#1]
Hi there. After upgrading to PHP 8.0, I get:

> PHP Fatal error:  Uncaught TypeError: decbin(): Argument #1 ($num) must be of type int, string given in /mybb/inc/plugins/antispam_by_cleantalk/lib/CleantalkAP/Common/Helper.php:255

I've been forced to patch lines 255 and 256 of the aforementioned file and cast the first argument:

```
$net_xtet = str_pad(decbin((int)($ip_type == 'v4' ? $net_ip_xtets[$xtet_count] : hexdec($net_ip_xtets[$xtet_count]))), $xtet_base, 0, STR_PAD_LEFT);
$ip_xtet = str_pad(decbin((int)($ip_type == 'v4' ? $ip_xtets[$xtet_count] : hexdec($ip_xtets[$xtet_count]))), $xtet_base, 0, STR_PAD_LEFT);
```

**Comment by svfcode** [comment]
Hello, @fvicent.

Please, give us a few days to inspect your case.

**Comment by svfcode** [comment]
Thank you for contacting us, we will fix the plugin within a couple of weeks.

**Comment by Glomberg** [comment]
The issue has been fixed in the 1.5 version.
